### DISCUSSION TITLE
Intercept the user endpoint response

### DISCUSSION THREAD

**Discussion by tiagomatosweb** [manchenkoff/nuxt-auth-sanctum#194] [Q&A] [answered]
Hi, I was just wondering if there is a way to interact/intercept the `endpoints.user` endpoint so that I can customise (right after the fetch) what should be available in the composable `useSanctumUser`.

For example, my `endpoints.user` endpoint returns 
```
{
id: 10,
name: 'Jon',
age: 30
}
```

But I want to make available in the composer only
```
{
id: 10,
name: 'Jon',
}
```

Removing the age.

Of course, it was just an hypothetical example.

Great plugin

**Comment by manchenkoff** [comment, marked as answer]
Hey @tiagomatosweb, there are two options depending on what you actually want to achieve:
1. `useSanctumUser` is a generic composable, so you can type it with a specific type/interface and your IDE will do the rest. However, you can still access other object properties even if they are not highlighted in the code. You can check example in the [docs](https://manchenkoff.gitbook.io/nuxt-auth-sanctum/usage/composables/usesantumuser)
2. You can create your custom global interceptor for ofetch `onResponse` and check if the request is your `endpoints.user`, then try to change `response` object. However, I never tried doing so, not sure how ofetch handles it. You can check another example in the [docs](https://manchenkoff.gitbook.io/nuxt-auth-sanctum/usage/interceptors) as well

**Comment by tiagomatosweb** [comment]
Using the global interceptor as you described on item 2 worked well.
Yeah legend! 🚀
